src/EventHub/EventHub.Autorest/custom/Set-AzEventHubNamespace.ps1

```diff
@@ -0,0 +1,345 @@
+        ${EncryptionKeyVaultProperty},
+
+        [Parameter(HelpMessage = "EncryptionRequireInfrastructureEncryption")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${EncryptionConfig}, #EncryptionConfig
+
+        [Parameter(HelpMessage = "IdentityType of EventHub Namespace.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${IdentityType},
+
+        [Parameter(HelpMessage = "IdentityUserAssigneIdentity")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.Collections.Hashtable]
+        ${IdentityId},#IdentityId
+
+        [Parameter(HelpMessage = "IsAutoInflateEnabled")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${IsAutoInflateEnabled},
+
+        [Parameter(HelpMessage = "KafkaEnabled")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${KafkaEnabled},
+
+        [Parameter(HelpMessage = "Location of EventHub Namespace")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${Location},
+
+        [Parameter(HelpMessage = "MaximumThroughputUnit of Eventhub Namespace.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.Int64]
+        ${MaximumThroughputUnit},
+
+        [Parameter(HelpMessage = "MinimumTlsVersion of Eventhub Namespace.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${MinimumTlsVersion},
+
+        [Parameter(HelpMessage = "PrivateEndpointConnectionName.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${PrivateEndpointConnection},
+
+        [Parameter(HelpMessage = "PublicNetworkAccess.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${PublicNetworkAccess},
+
+        [Parameter(HelpMessage = "The SkuCapacity of EventHub Namespace.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.Int64]
+        ${SkuCapacity},
+
+        [Parameter(HelpMessage = "The SkuName Of EventHub Namespace.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${SkuName},
+
+        [Parameter(HelpMessage = "SkuTier.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${SkuTier},
+
+        [Parameter(HelpMessage = "Tag of EventHub Namespace.")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.Collections.Hashtable]
+        ${Tag},
+
+        [Parameter(HelpMessage = "ZeroRedundant")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Path')]
+        [System.String]
+        ${ZoneRedundant},
```

Comment:
Remove "ZoneRedundant" as it is not an updatable property. Please correct type to SwitchParameter